spec/lib/dispatch-rider/logging/translator_spec.rb

```diff
@@ -0,0 +1,173 @@
+require 'spec_helper'
+
+describe DispatchRider::Logging::Translator do
+  let(:queue) { double :queue }
+  let(:message) { build(:message) }
+  let(:item) { double :item }
+  let(:fs_message) { DispatchRider::QueueServices::FileSystem::FsReceivedMessage.new(message, item, queue) }
+
+  let(:exception) { nil }
+  let(:duration) { nil }
+  let(:reason) { nil }
+
+  describe '.translate' do
+    # subject(:result) do
+    #   described_class.translate(message: fs_message, kind: kind, exception: exception, duration: duration, reason: reason)
+    # end
+
+    context 'when the job is starting' do
+      subject(:result) do
+        described_class.translate(fs_message, kind)
+      end
+      let(:kind) { :start }
+      let(:expected_hash) do
+        {
+          phase: :start,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the job has succeeded' do
+      subject(:result) do
+        described_class.translate(fs_message, kind)
+      end
+      let(:kind) { :success }
+      let(:expected_hash) do
+        {
+          phase: :success,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the job has failed' do
+      subject(:result) do
+        described_class.translate(fs_message, kind, exception: exception)
+      end
+      let(:kind) { :fail }
+      let(:exception) do
+        begin
+          raise(ArgumentError, "Foo is not bar")
+        rescue Exception => exception
```

Comment:
Oh, you just want an instance of an exception. :)

You don't need to do that. Just do:

```ruby
let(:exception) { ArgumentError.new "Foo is not bar" }
```